webpack/assets/javascripts/react_app/components/HostDetails/Templates/CardItem/CardTemplate/index.js

```diff
@@ -17,8 +21,21 @@ import {
   GridItem,
 } from '@patternfly/react-core';
 
-const CardItem = ({ content, header, children }) => {
+const CardItem = ({
+  content,
+  header,
+  children,
+  dropdownItems,
+  dropdownSelectHandler,
+}) => {
   const [activeAccordion, setActiveAccordion] = useState(0);
+  const [dropdownVisibility, setDropdownVisibility] = useState(false);
+
+  const onDropdownSelect = event => {
+    setDropdownVisibility(prevState => !prevState);
+    dropdownSelectHandler(event);
```

Comment:
Do we have a case where we would need an extra select handler? If not let's not add another prop to this component

Reply:
This is a common approach for templates components, this component is just a template that would be consumed in core and plugins.  This handler is needed when a consumer would like to invoke an action that related to a chosen item).

Reply:
I would much rather add additional props only when we have a case that requires them, instead of having a bunch YAGNI code and props cluttering the components.

Reply:
I have changed the approach here to be more generic.
Like overriding the wrapping Grid component, with the latest change consumers can override the Dropdown props.

There is no onClick callback per dropdown item (just href), so I wouldn't call this as YAGNI code, consumers can add their functionally only via onSelect callback.